lib/src/Base/Algo/openturns/CubaIntegration.hxx

```diff
@@ -0,0 +1,115 @@
+ *  it under the terms of the GNU Lesser General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  This library is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU Lesser General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Lesser General Public License
+ *  along with this library.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+#ifndef OPENTURNS_CUBAINTEGRATION_HXX
+#define OPENTURNS_CUBAINTEGRATION_HXX
+
+#include "openturns/IntegrationAlgorithmImplementation.hxx"
+
+BEGIN_NAMESPACE_OPENTURNS
+
+/**
+ * @class CubaIntegration
+ */
+
+class OT_API CubaIntegration
+  : public IntegrationAlgorithmImplementation
+{
+
+  CLASSNAME
+
+public:
+
+  /** Default constructor without parameters */
+  CubaIntegration();
+
+  /** Parameter constructor */
+  explicit CubaIntegration(const String optRoutine);
+
+  /** Virtual copy constructor */
+  CubaIntegration * clone() const override;
+
+  /* Compute an approximation of \int_{I}f(x)dx, where I is the integration interval */
+  using IntegrationAlgorithmImplementation::integrate;
+  Point integrate(const Function & function,
+                  const Interval & interval) const override;
+
+  /** maximumRelativeError accessor */
+  Scalar getMaximumRelativeError() const;
+  void setMaximumRelativeError(const Scalar maximumRelativeError);
+
+  /** maximumAbsoluteError accessor */
+  Scalar getMaximumAbsoluteError() const;
+  void setMaximumAbsoluteError(const Scalar maximumAbsoluteError);
+
+  /** optRoutine accessor */
+  String getOptRoutine() const;
+  void setOptRoutine(const String optRoutine);
+
+  /** maximumEvaluationNumber accessor */
+  UnsignedInteger getMaximumEvaluationNumber() const;
+  void setMaximumEvaluationNumber(const UnsignedInteger maximumEvaluationNumber);
+
+  /** flags accessor */
+  UnsignedInteger getFlags() const;
+  void setFlags(const UnsignedInteger flags);
+
+  /** String converter */
+  String __repr__() const override;
+
+  /** String converter */
+  String __str__(const String & offset = "") const override;
+
+  /** Method save() stores the object through the StorageManager */
+  void save(Advocate & adv) const override;
+
+  /** Method load() reloads the object from the StorageManager */
+  void load(Advocate & adv) override;
+private:
+
+  /* Friend C routine to compute the integrand function with parameter types
+     suitable for Cuba */
+  friend int computeIntegrand(const int *ndim, const double x[], const int *ncomp, double f[], void *userdata);
+
+  /* The number of dimensions of the integral */
+  UnsignedInteger nDim_ = 0;
+
+  /* The number of components of the integrand */
+  UnsignedInteger nComp_ = 0;
```

Comment:
are these actually used somewhere ?

Reply:
Oops no, these are legacy members from the development phase. Thanks for pointing this out!